autokeras/tuner.py

```diff
@@ -169,3 +172,194 @@ class RandomSearch(AutoTuner, kerastuner.RandomSearch):
+            if self._stage_trial_count == self._capacity[self._stage]:
+                self._stage = GreedyOracle.next_stage(self._stage)
+                self._stage_trial_count = 0
+            return {'status': kerastuner.engine.trial.TrialStatus.RUNNING,
+                    'values': values}
+        # All stages reached max collisions.
+        return {'status': kerastuner.engine.trial.TrialStatus.STOPPED,
+                'values': None}
+
+    def _generate_stage_values(self):
+        best_trials = self.get_best_trials()
+        if best_trials:
+            best_values = best_trials[0].hyperparameters.values
+        else:
+            best_values = self.hyperparameters.values
+        collisions = 0
+        while 1:
+            # Generate new values for the current stage.
+            values = {}
+            for p in self.hyperparameters.space:
+                if p.name in self._hp_names[self._stage]:
+                    values[p.name] = p.random_sample(self._seed_state)
+                    self._seed_state += 1
+            values = {**best_values, **values}
+            # Keep trying until the set of values is unique,
+            # or until we exit due to too many collisions.
+            values_hash = self._compute_values_hash(values)
+            if values_hash not in self._tried_so_far:
+                self._tried_so_far.add(values_hash)
+                break
+            collisions += 1
+            if collisions > self._max_collisions:
+                # Reached max collisions. No value to return.
+                return None
+        return values
+
+
+class Greedy(AutoTuner):
+
+    def __init__(self,
+                 hyper_graph,
+                 hypermodel,
+                 objective,
+                 max_trials,
+                 fit_on_val_data=False,
+                 seed=None,
+                 hyperparameters=None,
+                 tune_new_entries=True,
+                 allow_new_entries=True,
+                 **kwargs):
+        self.seed = seed
+        oracle = GreedyOracle(
+            hyper_graph=hyper_graph,
+            objective=objective,
+            max_trials=max_trials,
+            seed=seed,
+            hyperparameters=hyperparameters,
+            tune_new_entries=tune_new_entries,
+            allow_new_entries=allow_new_entries)
+        hp = oracle.get_space()
+        preprocess_graph, keras_graph = hyper_graph.build_graphs(hp)
+        oracle.update_space(hp)
+        super().__init__(
+            hyper_graph=hyper_graph,
+            fit_on_val_data=fit_on_val_data,
+            oracle=oracle,
+            hypermodel=hypermodel,
+            **kwargs)
+
+
+TUNER_CLASSES = {
+    'random_search': RandomSearch,
+    'greedy': Greedy,
+    'image_classifier': Greedy,
+    'image_regressor': Greedy,
+    'text_classifier': Greedy,
+    'text_regressor': Greedy,
+    'structured_data_classifier': Greedy,
+    'structured_data_regressor': Greedy,
+}
+
+
+def get_tuner_class(name):
```

Comment:
This needs to be extended to accept passing a tuner instance rather than a string.

Also, if an invalid value is passed, the error message should be informative and actionable, saying, "the value {value} passed for argument `tuner` is invalid, expected one of {list of values}"